prime_directive/core/tmux.py

```
@@ -1,98 +1,105 @@
-        # We do NOT put a timeout here as it would kill the user's session.
-        if attach:
-            subprocess.run(["tmux", "attach-session", "-t", session_name])
+            await asyncio.wait_for(proc.wait(), timeout=2.0)
+        except asyncio.TimeoutError:
+            logger.error("tmux switch-client timed out")
+    elif attach:
+        # attach-session is interactive and blocks until the user detaches —
+        # run it without a timeout via a standard subprocess to preserve interactivity.
+        import subprocess
 
+        subprocess.run(["tmux", "attach-session", "-t", session_name])
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)